### PR TITLE
Docs: Split up Build Notifications into feature/reference and how-to (Diátaxis)

### PR DESCRIPTION
**Pull request by benjaoming**
This made a lot of sense.

I had to write new text for the old feature description, though. But I think that text was already missing in the former version.

### CODE REVIEW

File: docs/user/build-notifications.rst [hunk 1]
```diff
@@ -1,278 +1,24 @@
...
 
-Take these steps to enable build notifications using email:
+  Webhooks can be customized through your own template and a variety of variable substitutions.
 
-* Go to :guilabel:`Admin` > :guilabel:`Notifications` in your project.
-* Fill in the **Email** field under the **New Email Notifications** heading
-* Submit the form
+.. seealso::
+    :doc:`/guides/build-notifications`
+        All the practical steps for setting up build notifications
```
Comment (benjaoming) [inline docs/user/build-notifications.rst:21]: The above added lines are all new texts.

File: docs/user/build-notifications.rst [hunk 2]
```diff
@@ -1,278 +1,24 @@
...
-This makes sure you know when your builds have failed.
+  Slack and Discord are supported through ready-made templates.
 
-Take these steps to enable build notifications using email:
+  Webhooks can be customized through your own template and a variety of variable substitutions.
 
-* Go to :guilabel:`Admin` > :guilabel:`Notifications` in your project.
-* Fill in the **Email** field under the **New Email Notifications** heading
-* Submit the form
+.. seealso::
```
Comment (ericholscher) [inline docs/user/build-notifications.rst:19]: This is a good start to a baseline feature landing page. I'd probably link to the Slack & Discord templates, and perhaps have a screenshot or something here?

**Review by ericholscher — APPROVED** [review]
This is a better refactor. I think we can continue to build out the feature pages, but at least this has one 👍

File: docs/user/build-notifications.rst (hunk 2 above)
Comment (benjaoming) [reply docs/user/build-notifications.rst:19]: Noted for subsequent updates iteration :+1:

**Pull request merged**